exercises/affine-cipher/.meta/generator/affine_cipher_case.rb

```diff
@@ -0,0 +1,51 @@
+require 'generator/exercise_case'
+
+class AffineCipherCase < Generator::ExerciseCase
+
+  def workload
+    return error_workload if error_expected?
+    case property
+    when 'encode' then encode_workload
+    when 'decode' then decode_workload
+    else raise 'unexpected property encountered'
+    end
+  end
+
+  private
+
+  def error_expected?
+    Hash === expected && expected.key?('error')
+  end
+
+  def encode_workload
+    key_a, key_b = input['key']['a'], input['key']['b']
+    indent_lines(
+      [
+        "cipher = Affine.new(#{key_a}, #{key_b})",
+        "plaintext = '#{input['phrase']}'",
+        "ciphertext = '#{expected}'",
+        "assert_equal ciphertext, cipher.encode(plaintext)"
+      ], 4
+    )
+  end
+
+  def decode_workload
+    key_a, key_b = input['key']['a'], input['key']['b']
+    indent_lines(
+      [
+        "cipher = Affine.new(#{key_a}, #{key_b})",
+        "plaintext = '#{expected}'",
+        "ciphertext = '#{input['phrase']}'",
+        "assert_equal plaintext, cipher.decode(ciphertext)"
+      ], 4
+    )
+  end
+
+  def error_workload
+    key_a, key_b = input['key']['a'], input['key']['b']
```

Comment:
Pro tip: 
Each of these methods contains a `key_a, key_b = input['key']['a'], input['key']['b']` line.

I'd extract 2 methods that look something like:
```
def key_a
  input['key']['a']
end
```

This way it's clear 
a) what key_a is.
b) it's the same for all tests.
c) you're not repeating yourself all over the place.
d) if the key used in the canonical data ever changes (low probability, but possible) you only need to fix it in one place.